goth/configuration.py

```diff
@@ -252,3 +272,16 @@ def load_yaml(yaml_path: Union[Path, str]) -> Configuration:
             )
 
     return config
+
+
+def _apply_overrides(dict_: Dict[str, Any], overrides: List[Override]):
+    for (dict_path, value) in overrides:
+        path_list: List[str] = dict_path.split(".")
```

Comment:
I think you could also allow `int`s in `path_list` and then override like this:
```
    overrides = [("nodes.1.name", "dostawca-1")]  # path_list = ["nodes", 1, "name"]
    config = load_yaml(default_config_file, overrides)
    assert config.containers[1].name == "dostawca-1"
```
But it's probably not worth doing, if we ever need to identify config elements based on list index then it would be probably an indication that the relevant part of the configuration tree should be made a `dir` instead of a `list`.

Reply:
I think it's better to limit ourselves to `dict`s for now. The only use-cases I see for list access in our current config file is for debugging purposes. As such, this can be achieved by editing the file itself, rather than using overrides.
With that said I'm going to mention the limitation to `dict` in the docstring for `load_yaml`.

Reply:
Added in 8f83b54a9027728950ccda1b59a1bae864bf5270